File: hash/programmers_42576/solution.js
function solution(participant, completion) {
    const participantMap = new Map()
    
    for (const p of participant) {
        participantMap.set(p, (participantMap.get(p) || 0) + 1)
    }
    
    for (const c of completion) {
        participantMap.set(c, participantMap.get(c) - 1);
    }
    
    for (const [name, count] of participantMap) {
        if(count) {
            return name
        }
    }
}

File: hash/programmers_42576/solution.py
def solution(participant, completion):
    hash_table = {}
    
    for p in participant:
        hash_table[p] = hash_table.get(p, 0) + 1
        
    for c in completion:
        hash_table[c] -= 1
    
    for name, count in hash_table.items():
        if count:
            return name